sdks/java/io/snowflake/src/main/java/org/apache/beam/sdk/io/snowflake/services/SnowflakeServices.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.snowflake.services;
+
+import java.io.Serializable;
+
+/** Interface which defines common methods for interacting with Snowflake. */
+public interface SnowflakeServices extends Serializable {
```

Comment:
Is there any chance the `SnowflakeService` -> `SnowflakeServices` rename will be a breaking change for our users? They are public after all

Reply:
This class has been marked public, but should not be used in public because it touches upon very internal things. I have now added annotations to mark these methods as only available for tests.
I think we shouldn't be concerned about the backward compatibility of this class any more than it does when making a change to BigQueryServices.
https://github.com/apache/beam/commits/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServices.java

Reply:
@TheNeuralBit What do you think?

Reply:
I'd suggest just annotating these classes with `@Internal` (I'm surprised we didn't do this in `BigQueryServices` already)